src/Components/SourcesSelect/index.js

```diff
@@ -24,9 +24,18 @@ const SourcesSelect = ({ setValidation }) => {
     enabled: !!provider,
     onSuccess: (data) => {
       const id = chosenSource;
-
-      if (!id) return;
-      setSelected(selectObject(id, data.find((source) => source.id === id).name));
+      if (provider === AWS_PROVIDER) {
+        setSelected(selectObject(imageSourceID, data.find((source) => source.id === imageSourceID).name));
```

Comment:
The error here is that source.id (comming from the mocked response) is an integer, but the imageSourceID is a string. I guess `==` would fix this as it would allow type mismatch :)

Reply:
didn't help :(